packages/dev/core/src/Materials/PBR/pbrBaseMaterial.ts

```diff
@@ -2298,11 +2298,10 @@ export abstract class PBRBaseMaterial extends PushMaterial {
                 if (defines.METALLICWORKFLOW) {
                     TmpColors.Color3[0].r = this._metallic === undefined || this._metallic === null ? 1 : this._metallic;
                     TmpColors.Color3[0].g = this._roughness === undefined || this._roughness === null ? 1 : this._roughness;
-                    ubo.updateColor4("vReflectivityColor", TmpColors.Color3[0], 1);
-
                     const ior = this.subSurface?._indexOfRefraction ?? 1.5;
                     const outsideIOR = 1; // consider air as clear coat and other layers would remap in the shader.
-
+                    TmpColors.Color3[0].b = ior;

```

Comment:
Here, I'm putting the IOR value into the third channel of the `vReflectivityColor`. Why does the uniform have this name if it's not a colour?
And `vMetallicReflectanceFactors` stores the F0 colour but uses a name with "Factors" in it...
And several of these uniforms are preceded by the word "metallic" when they seem to only apply to dielectrics...

Reply:
Oh, maybe they're only used in the metallic workflow and that's why they have the word "metallic" as a prefix?